lib/node.js

```diff
@@ -1407,7 +1407,9 @@ declare class stream$Stream extends events$EventEmitter {}
 
 type readableStreamOptions = { highWaterMark? : number, encoding? : ?string, objectMode? : boolean };
 declare class stream$Readable extends stream$Stream {
+  _destroy(error: Error, callback: () => void): void;
```

Comment:
The docs mention 

> callback <Function> A callback function that takes an optional error argument.

Could you add this optional error argument to the callback?

Reply:
Sure thing. It should become the following, right?

  `_destroy(error: Error, callback: (error? : Error) => void): void;`

Reply:
`error` is also optional. Maybe the best option is `_destroy(error: ?Error, callback?: (error?: ?Error) => void): void;`

Reply:
This is how they documented that type in `@types/node`.

![types_node](https://user-images.githubusercontent.com/12381818/46042328-64896500-c115-11e8-9309-dbc80c04078c.png)

That implies that neither `writable._destroy();` or `writable._destroy(null);` are valid in Typescript at the moment, as the two parameters are required.
In fact, a quick test reveals that they are right. If I try to compile the following Typescript source, I get an error.

```typescript
import * as stream from 'stream';
const writable = new stream.Writable();
// @ts-ignore
writable._destroy(); // this throws
// writable._destroy(null, () => { /* */ }); (this wouldn't throw)
```

```
_stream_writable.js:673
  cb(err);
  ^

TypeError: cb is not a function
    at Writable._destroy (_stream_writable.js:673:3)
    at Object.<anonymous> (C:\Work\test\index.js:4:10)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:612:3
```

Reply:
I suspect the TypeScript definition is wrong, then. The [documentation for stream](https://nodejs.org/dist/latest-v8.x/docs/api/stream.html#stream_writable_destroy_err_callback) states that `error` is a _possible_ `Error`:

Reply:
@DullReferenceException With my example I wanted to demonstrate that `@types/node` isn't wrong, and that my implementation is based on top of it.

My typescript example compiled to javascript with `tsc` gives the following:

```
const stream = __importStar(require("stream"));
const writable = new stream.Writable();
// writable._destroy(null, () => { /* */ });
// @ts-ignore
writable._destroy();
//# sourceMappingURL=index.js.map
```

The `// @ts-ignore` directive simply forced Typescript to compile even when if fails to type check.
The error is thrown from the NodeJS core itself. The documentation states that the `err` parameter is a *possible* `Error`, meaning that it can also be `null`. But you can't omit the two params, since there isn't any check that tests wether `cb` (the callback with an optional error) is `null` / `undefined` or not.

Reply:
Yes, you are correct. New PR incoming. The first error can in fact accept 3 alternatives:
1) an `Error` instance
2) `null`
3) `undefined`

I just didn't remember the exact syntax in Flow (i.e. inverting the `:` and the `?` symbols).
Thanks for the refresh.